pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -2708,10 +2594,10 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         using ComplexT = std::complex<PrecisionT>;
         constexpr ComplexT imag{0, 1};
         auto core_function =
-            [imag](std::complex<PrecisionT> *arr,
-                   [[maybe_unused]] const std::size_t i0011,
-                   [[maybe_unused]] const std::size_t i1100,
-                   [[maybe_unused]] const std::vector<std::size_t> &indices) {
+            [imag](std::complex<PrecisionT> *arr, const std::size_t i0011,
```

Comment:
We can exclude constexpr values in the lambda clause.
```suggestion
            [](std::complex<PrecisionT> *arr, const std::size_t i0011,
```